src/compiler/lowering/SpecializeGenericFunctionsPass.cpp

```diff
@@ -223,6 +239,46 @@ namespace {
                     specializeCallsInFunction(calledFunction);
                 }
             });
+
+            // Specialize all functions called by MapOp
+            function.walk([&](daphne::MapOp mapOp) {
+                auto calledFunction = functions[mapOp.func().str()];
```

Comment:
`functions` contains `FuncOp` by their symbol names (i.e. the ones created by the parser). To look for other `FuncOps`with the same identifier (e.g. for the purpose mentioned in the remark below), one would need to implement an appropriate search through the keys of `functions`.
The problem then is, that there might be different `FuncOp`s that fit the specialization and one would need to decide which one to prefer.